key.core/src/main/java/de/uka/ilkd/key/proof/mgt/SpecificationRepository.java

```diff
@@ -1847,4 +1850,87 @@ public ImmutableSet<WellDefinednessCheck> getAllWdChecks() {
         }
         return result;
     }
+
+
+    // region Support SetStatement and JmlAssert
+    private final Map<Statement, JmlStatementSpec> statementMap = new IdentityHashMap<>();
+
+    @Nullable
+    public JmlStatementSpec getStatementSpec(Statement statement) {
+        return statementMap.get(statement);
+    }
+
+    public JmlStatementSpec addStatementSpec(Statement statement, JmlStatementSpec spec) {
+        return statementMap.put(statement, spec);
+    }
+
+    /**
+     * This record represents mutable information for JML statements.
+     * JML statements need to maintain the current variable set as well as the updated information for the KeY terms
+     * they describe.
+     * <p>
+     * For {@link de.uka.ilkd.key.java.statement.JmlAssert} this is the formula behind the assert.
+     * For {@link de.uka.ilkd.key.java.statement.SetStatement} this is the target and the value terms.
```

Comment:
that's a bit unclear. why is it mutable? in a record?
What is "This is"? Would that not always be one formula/term at most? Why collections?
That would make nasty bugs otherwise. There was something with loops specs once ...

Reply:
I wrote a little bit more.